contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,312 @@
+        uint256 newWeth = wethBalance - depositedWethAccountedFor;
+
+        if (newWeth > 0) {
+            depositedWethAccountedFor = wethBalance;
+
+            _deposit(WETH, newWeth);
+        }
+    }
+
+    /// @notice Withdraw WETH from this contract. Used only if some WETH for is lingering on the contract. That
+    /// can happen when:
+    ///   - the deposit was not a multiple of 32 WETH
+    ///   - someone sent WETH directly to this contract
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    /// @param _recipient Address to receive withdrawn assets
+    /// @param _asset WETH to withdraw
+    /// @param _amount Amount of WETH to withdraw
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        _withdraw(_recipient, _asset, _amount);
+    }
+
+    function _withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+
+        emit Withdrawal(_asset, address(0), _amount);
+        IERC20(_asset).safeTransfer(_recipient, _amount);
+    }
+
+    /// @notice transfer all WETH deposits back to the vault.
+    /// This does not withdraw from the validators. That has to be done separately with the
+    /// `exitSsvValidator` and `removeSsvValidator` operations.
+    /// This does not withdraw any execution rewards from the FeeAccumulator or
+    /// consensus rewards in this strategy.
+    /// Any ETH in this strategy that was swept from a full validator withdrawal will not be withdrawn.
+    /// ETH from full validator withdrawals is sent to the Vault using `doAccounting`.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    function withdrawAll() external override onlyVaultOrGovernor nonReentrant {
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (wethBalance > 0) {
+            _withdraw(vaultAddress, WETH, wethBalance);
+        }
+    }
+
+    /// @notice Returns the total value of (W)ETH that is staked to the validators
+    /// and WETH deposits that are still to be staked.
+    /// This does not include ETH from consensus rewards sitting in this strategy
+    /// or ETH from MEV rewards in the FeeAccumulator. These rewards are harvested
+    /// and sent to the Dripper so will eventually be sent to the Vault as WETH.
+    /// @param _asset      Address of weth asset
+    /// @return balance    Total value of (W)ETH
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256 balance)
+    {
+        require(_asset == WETH, "Unsupported asset");
+
+        balance =
+            // add the ETH that has been staked in validators
+            activeDepositedValidators *
+            32 ether +
```

Comment:
We use `32 eth` several times in this file. Perhaps this should change to FULL_STAKE?

Reply:
good idea as EIP-7251, Increase the MAX_EFFECTIVE_BALANCE, will allow us to stake more than 32 ether.